### PR TITLE
EFI-166

### PR DESCRIPTION
**Pull request by martinb89**

### CODE REVIEW

File: SubProjects/MessageModel/MessageModel/MessageModel/Interface/FetchImap/FetchImapFoldersService.swift [hunk 1]
```diff
@@ -76,6 +76,16 @@ public class FetchImapFoldersService {
                       alsoCreatePEPFolder: Bool = false,
                       saveContextWhenDone: Bool = true,
                       completion: @escaping CompletionBlock) {
+        func addOperationAndWait(blockingQueue: OperationQueue, operation: ImapSyncOperation) {
```
Comment (ghost) [inline SubProjects/MessageModel/MessageModel/MessageModel/Interface/FetchImap/FetchImapFoldersService.swift:79]: I really hope this extra blocking code is not needed. Why now?

File: SubProjects/MessageModel/MessageModel/MessageModel/Interface/FetchImap/FetchImapFoldersService.swift [hunk 2]
```diff
@@ -76,6 +76,16 @@ public class FetchImapFoldersService {
...
+            blockingQueue.addOperation(operation)
+            blockingQueue.waitUntilAllOperationsAreFinished()
+        }
+
+        func addFolderCreationOperation(context: NSManagedObjectContext?, imapConnection: ImapConnection, blockingQueue: OperationQueue, folderType: FolderType) {
```
Comment (ghost) [inline SubProjects/MessageModel/MessageModel/MessageModel/Interface/FetchImap/FetchImapFoldersService.swift:84]: Add folders while fetching them? That doesn't sound right.

**Comment by ghost** [comment]
Apart from the UI changes, the backend changes look way too much and complicated. Did we introduce bugs in the "special folders creation" that we never detected?

I was expecting a change in the create folders operation, an addition to the corresponding service for handling the other folders, plus a new constant.

What do we do here? Shall we split the ticket in the backend part (creating the special part) and the actual logic?

**Review by ghost — CHANGES_REQUESTED** [review]

File: SubProjects/MessageModel/MessageModel/MessageModel/Interface/FetchImap/FetchImapFoldersService.swift (hunk 2 above)
Comment (martinb89) [reply SubProjects/MessageModel/MessageModel/MessageModel/Interface/FetchImap/FetchImapFoldersService.swift:84]: Actually this is what fixes what we discuss yesterday. 
Here we create planck folder as well.

File: SubProjects/MessageModel/MessageModel/MessageModel/Interface/FetchImap/FetchImapFoldersService.swift (hunk 1 above)
Comment (martinb89) [reply SubProjects/MessageModel/MessageModel/MessageModel/Interface/FetchImap/FetchImapFoldersService.swift:79]: It's used. I just extracted duplicated code.

**Review by ghost — APPROVED** [review]

**Pull request merged**